src/FluentDriver.php

```diff
@@ -117,6 +127,103 @@ public function getMappers()
     }
 
     /**
+     * Add mappings from an array of folders.
+     *
+     * @param string[] $paths
+     *
+     * @throws MappingException
+     */
+    public function addPaths($paths)
+    {
+        foreach ($paths as $path) {
+            if (!is_dir($path)) {
+                throw MappingException::fileMappingDriversRequireConfiguredDirectoryPath($path);
+            }
+
+            $iterator = new \RecursiveIteratorIterator(
+                new \RecursiveDirectoryIterator($path),
+                \RecursiveIteratorIterator::LEAVES_ONLY
+            );
+
+            foreach ($iterator as $file) {
+                if ($file->getBasename('.php') == $file->getBasename()) {
+                    continue;
+                }
+
+                $sourceFile = realpath($file->getPathName());
+                $className = $this->getClassFromFile($sourceFile);
+                $this->addMapping(new $className());
+            }
+        }
+    }
+
+    /**
+     * Get the FQN of a class from a source file.
+     *
+     * @param $pathToFile
+     *
+     * @return string
+     */
+    private function getClassFromFile($pathToFile)
+    {
+        //http://jarretbyrne.com/2015/06/197/
+        //Grab the contents of the file
+        $contents = file_get_contents($pathToFile);
+
+        //Start with a blank namespace and class
+        $namespace = $class = '';
+
+        //Set helper values to know that we have found the namespace/class token and need to collect the string values after them
+        $getting_namespace = $getting_class = false;
+
+        //Go through each token and evaluate it as necessary
+        foreach (token_get_all($contents) as $token) {
```

Comment:
@FoxxMD this is why I didn't add the file parser right away. Tokenizing a file's contents seems overkill, when the alternative is to list the names in the config and be done with it.
I'm not comfortable with the current solution, I don't know what @patrickbrouwers thinks about it.